packages-content-model/roosterjs-content-model-core/lib/publicApi/selection/deleteSelection.ts

```diff
@@ -13,27 +14,28 @@ import type {
  * @param model The model to delete selected content from
  * @param additionalSteps @optional Addition delete steps
  * @param formatContext @optional A context object provided by formatContentModel API
+ * @param additionalStepsResult  @optional The delete result to trigger the additional steps @default 'notDeleted'
  * @returns A DeleteSelectionResult object to specify the deletion result
  */
 export function deleteSelection(
     model: ContentModelDocument,
     additionalSteps: (DeleteSelectionStep | null)[] = [],
-    formatContext?: FormatWithContentModelContext
+    formatContext?: FormatWithContentModelContext,
+    additionalStepsResult: DeleteResult = 'notDeleted'
```

Comment:
I can understand you need tot execute the step when delete result is range. But this parameter is not easy to understand.

I think we can remove the check at line 32, and move it into all existing delete steps to do this check there. So that you can check the delete result to be 'range' in your function directly.